src/main/java/io/datareplication/internal/page/PageLoader.java

```diff
@@ -0,0 +1,98 @@
+package io.datareplication.internal.page;
+
+import io.datareplication.consumer.HttpException;
+import io.datareplication.consumer.PageFormatException;
+import io.datareplication.consumer.StreamingPage;
+import io.datareplication.internal.http.HttpClient;
+import io.datareplication.internal.multipart.BufferingMultipartParser;
+import io.datareplication.internal.multipart.MultipartException;
+import io.datareplication.internal.multipart.MultipartParser;
+import io.datareplication.model.HttpHeader;
+import io.datareplication.model.HttpHeaders;
+import io.datareplication.model.Url;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.Maybe;
+import io.reactivex.rxjava3.core.Single;
+import lombok.NonNull;
+import org.reactivestreams.FlowAdapters;
+
+import java.io.IOException;
+import java.net.http.HttpResponse;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.Flow;
+import java.util.stream.Stream;
+
+public class PageLoader {
+    private final HttpClient httpClient;
+
+    public PageLoader(final HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    public Single<StreamingPage<HttpHeaders, HttpHeaders>> load(Url url) {
```

Comment:
It's internal, but it throws a runtimeException, this should be commented (JavaDoc)